drivers/sensors/pmw3389.c

```diff
@@ -0,0 +1,289 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pmw3389.h"
+#include "wait.h"
+#include "debug.h"
+#include "print.h"
+#include PMW3389_FIRMWARE_H
+
+// Registers
+// clang-format off
+#define REG_Product_ID                 0x00
+#define REG_Revision_ID                0x01
+#define REG_Motion                     0x02
+#define REG_Delta_X_L                  0x03
+#define REG_Delta_X_H                  0x04
+#define REG_Delta_Y_L                  0x05
+#define REG_Delta_Y_H                  0x06
+#define REG_SQUAL                      0x07
+#define REG_RawData_Sum                0x08
+#define REG_Maximum_RawData            0x09
+#define REG_Minimum_RawData            0x0A
+#define REG_Shutter_Lower              0x0B
+#define REG_Shutter_Upper              0x0C
+#define REG_Ripple_Control             0x0D
+#define REG_Resolution_L               0x0E
+#define REG_Resolution_H               0x0F
+#define REG_Config2                    0x10
+#define REG_Angle_Tune                 0x11
+#define REG_Frame_Capture              0x12
+#define REG_SROM_Enable                0x13
+#define REG_Run_Downshift              0x14
+#define REG_Rest1_Rate_Lower           0x15
+#define REG_Rest1_Rate_Upper           0x16
+#define REG_Rest1_Downshift            0x17
+#define REG_Rest2_Rate_Lower           0x18
+#define REG_Rest2_Rate_Upper           0x19
+#define REG_Rest2_Downshift            0x1A
+#define REG_Rest3_Rate_Lower           0x1B
+#define REG_Rest3_Rate_Upper           0x1C
+#define REG_Observation                0x24
+#define REG_Data_Out_Lower             0x25
+#define REG_Data_Out_Upper             0x26
+#define REG_SROM_ID                    0x2A
+#define REG_Min_SQ_Run                 0x2B
+#define REG_RawData_Threshold          0x2C
+#define REG_Control2                   0x2D
+#define REG_Config5_L                  0x2E
+#define REG_Config5_H                  0x2F
+#define REG_Power_Up_Reset             0X3A
+#define REG_Shutdown                   0x3B
+#define REG_Inverse_Product_ID         0x3F
+#define REG_LiftCutoff_Cal3            0x41
+#define REG_Angle_Snap                 0x42
+#define REG_LiftCutoff_Cal1            0x4A
+#define REG_Motion_Burst               0x50
+#define REG_SROM_Load_Burst            0x62
+#define REG_Lift_Config                0x63
+#define REG_RawData_Burst              0x64
+#define REG_LiftCutoff_Cal2            0x65
+#define REG_LiftCutoff_Cal_Timeout     0x71
+#define REG_LiftCutoff_Cal_Min_Length  0x72
+#define REG_PWM_Period_Cnt             0x73
+#define REG_PWM_Width_Cnt              0x74
+
+// clang-format on
+
+#ifndef MAX_CPI
+#    define MAX_CPI 0x013F  // limits to 0--319, should be max cpi/50 -1
```

Comment:
I found this comment very cryptic and have changed it for pmw3360 in https://github.com/uqs/qmk_firmware/commit/ddca5df530b17a36ac22ea0ea447e8244bd2d025

Did you find then min_cpi that the sensor supports? Can we assume its 50?

Reply:
as you mentioned nothing in the data sheet explicitly states the min cpi setting I guess we can infer that it is 50 I will do some tests to confirm with this along with testing the tSRAD addition (see below)
I agree with improving the comment I added one that follows your format